packages/type-compiler/tests/inline-external-imports.spec.ts

```diff
@@ -0,0 +1,456 @@
+                  "parent": [Circular],
+                },
+                "typeName": undefined,
+              },
+              "type": {
+                "kind": 17,
+                "name": undefined,
+                "parameters": [
+                  {
+                    "kind": 18,
+                    "name": "value",
+                    "parent": [Circular],
+                    "type": {
+                      "kind": 1,
+                      "parent": [Circular],
+                    },
+                  },
+                  {
+                    "kind": 18,
+                    "name": "index",
+                    "parent": [Circular],
+                    "type": {
+                      "kind": 6,
+                      "parent": [Circular],
+                    },
+                  },
+                ],
+                "parent": [Circular],
+                "return": {
+                  "kind": 1,
+                  "parent": [Circular],
+                },
+              },
+            },
+          ],
+          "return": {
+            "kind": 0,
+            "parent": {
+              "function": [Function],
+              "kind": 17,
+              "name": "map",
+              "parameters": [
+                {
+                  "kind": 18,
+                  "name": "project",
+                  "parent": [Circular],
+                  "type": {
+                    "kind": 17,
+                    "name": undefined,
+                    "parameters": [
+                      {
+                        "kind": 18,
+                        "name": "value",
+                        "parent": [Circular],
+                        "type": {
+                          "kind": 1,
+                          "parent": [Circular],
+                        },
+                      },
+                      {
+                        "kind": 18,
+                        "name": "index",
+                        "parent": [Circular],
+                        "type": {
+                          "kind": 6,
+                          "parent": [Circular],
+                        },
+                      },
+                    ],
+                    "parent": [Circular],
+                    "return": {
+                      "kind": 1,
+                      "parent": [Circular],
+                    },
+                  },
+                },
+              ],
+              "return": [Circular],
+              "typeName": undefined,
+            },
+          },
+          "typeName": undefined,
+        }
+    `);
+})
+
+test('class type var', () => {
+    const res = transpile({
+        app: `import { Observable } from 'rxjs';
+
+            type A = Observable<unknown>;
+        `
+    }, undefined, {
+        inlineExternalLibraryImports: {
+            'rxjs': ['Observable'],
+        },
+    });
+
+    // TODO: how to resolve the typeof imports? -> __assignType(Subscriber_1.Subscriber, __ΩSubscriber)
```

Comment:
@marcj any suggestion on how we might solve this?

Reply:
we might also have an issue with duplicated variables...
If X is specified in the source file, and a type named X is also used in the external import type, then they'll clash

Reply:
we could perhaps prefix the variable names with something extra to differentiate

Reply:
the `e` that angular uses?

Reply:
`__eΩSubscriber`

Reply:
example? Dynamically rename is an idea. we have all symbol names in the AST to check for that

Reply:
alternative is to use scoped variables, so basically use const instead of var and put tehm into their own {}. this way we can encapsulate each embedded module correctly

Reply:
@marcj yeah, I think the scoped variables might be a good idea for encapsulation embedded modules correctly.
Currently, my solution is to prefix external runtime type names with `__ɵΩ` instead of `__Ω`.

Reply:
```ts
const __ɵΩrxjs = {
  Observable: [],
  Subscription: [],
  // etc...
};
```

Reply:
@marcj what should we do about function & class references? I've thought about converting class references to interfaces and function references to function types.

Reply:
Otherwise we could make two new types
`ExternalTypeClass` and `ExternalTypeFunction` which do not have class/function references.

Reply:
or you import the real thing and adjust their __type with the actual program so it behaves like normal class/function in deepkit runtime types